imap/jmap_mail_query.c

```diff
@@ -250,27 +294,47 @@ HIDDEN int jmap_email_contactfilter_from_filtercondition(struct jmap_parser *par
         if (!groupid) continue;
         if (hash_lookup(groupid, &cfilter->contactgroups)) continue;
 
-        /* Lookup group member email addresses */
-        mbentry_t *mbentry = NULL;
-        strarray_t *members = NULL;
-        if (!cfilter->addrbook ||
-            !mboxlist_lookup(cfilter->addrbook, &mbentry, NULL)) {
-            members = carddav_getemails(cfilter->carddavdb, mbentry, groupid,
-                                        (c->type == CF_GROUP) ?
-                                        CARDDAV_KIND_GROUP : CARDDAV_KIND_ANY,
-                                        othermb);
+        /* Lookup card email addresses and any group members */
+        unsigned card_kind =
+            (c->type == CF_GROUP) ? CARDDAV_KIND_GROUP : CARDDAV_KIND_ANY;
+        strarray_t card_uids = { 1, 0, (char **) &groupid };
```

Comment:
This looks unnecessarily dangerous. I guess it's done to avoid duplicating the groupid string, but it's tied close to internals of the strarray implementation that the API does not guarantee. I'd say using `STRARRAY_INITIALIZER` and regularly appending the string is OK here, the remaining code in this loop already costs heap allocations.